### PR TITLE
Implement header-body split (make block-fetch actually work) in diffusion tests

### LINKED ISSUE #4389: Finish header-body split in net-sim
Currently `chain-sync` is using blocks rather than headers, we also need to make sure that block-fetch is requesting blocks when their headers are announced through `chain-sync`.

We have an [example](https://github.com/input-output-hk/ouroboros-network/blob/master/ouroboros-network/demo/chain-sync.hs) which might be helpful.

We should add a label which shows how many blocks a block-fetch client downloaded over the simulation.

### PR DESCRIPTION
**Pull request by bolt12**
Closes #4389

### CODE REVIEW

File: ouroboros-network/test/Test/Ouroboros/Network/Testnet/Simulation/Node.hs [hunk 1]
```diff
@@ -286,27 +286,21 @@ fixupCommands (jn@(JoinNetwork _ _):t) = jn : go jn t
...
   -- the IOSim.
   --
   -- Quota values matches mainnet, so a slot length of 1s and 1 / 20
   -- chance that someone gets to make a block
   let bgaSlotDuration = secondsToDiffTime 1
-      numberOfNodes   = length [ r | r@(RelayAccessAddress _ _) <- raps ]
-      quota = if numberOfNodes > 0
-                then 20 `div` numberOfNodes
-                else 100
+      quota = 5 -- 1 / 20 * 100
```
[the remark this answers is not in the document]
Comment (bolt12) [reply ouroboros-network/test/Test/Ouroboros/Network/Testnet/Simulation/Node.hs:299]: I forgot to change this to the original value, I was doing experiments to check if block-fetch was working!

File: ouroboros-network/test/Test/Ouroboros/Network/Testnet.hs [hunk 2]
```diff
@@ -316,6 +320,42 @@ prop_inbound_governor_transitions_coverage defaultBearerInfo diffScript =
...
+      events = mapMaybe (\case DiffusionFetchTrace st ->
+                                    Just st
+                               _ -> Nothing
+                        )
+             . Trace.toList
+             . fmap (\(WithTime _ (WithName _ b)) -> b)
+             . withTimeNameTraceEvents
+                @DiffusionTestTrace
+                @NtNAddr
+             . Trace.fromList (MainReturn (Time 0) () [])
```
[the remark this answers is not in the document]
Comment (bolt12) [reply ouroboros-network/test/Test/Ouroboros/Network/Testnet.hs:349]: There's no easy way to just operate on a single type. The trace comes in a different type and if we only want to analyze 125k worth of traces the easiest way is to convert to a list an `take` .

[the remark this answers is not in the document]
Comment (coot) [reply ouroboros-network/test/Test/Ouroboros/Network/Testnet.hs:349]: Right, my question was if we can operate on a list without using `Trace` by avoiding the `toList`.

[the remark this answers is not in the document]
Comment (bolt12) [reply ouroboros-network/test/Test/Ouroboros/Network/Testnet.hs:349]: I mean we probably can, but it is more convenient to use lists here imo. It should get fused away no?

**Comment by bolt12** [comment]
bump

**Review by coot — APPROVED** [review]
LGTM

**Comment by bolt12** [comment]
bors merge

**Pull request merged**